runtime/src/authorize_and_stash.rs

```diff
@@ -62,20 +63,52 @@ impl AuthorizeAndStashCmd {
             // Check if image hash is present in the image metadata entry collection.
             let persistent_data = drivers.persistent_data.get();
             let auth_manifest_image_metadata_col =
-                &persistent_data.auth_manifest_image_metadata_col;
-
-            let mut auth_result = DENY_IMAGE_AUTHORIZATION;
-            for metadata_entry in auth_manifest_image_metadata_col.image_metadata_list.iter() {
-                if cfi_launder(metadata_entry.digest) == cmd.measurement {
-                    caliptra_cfi_lib_git::cfi_assert_eq_12_words(
-                        &Array4x12::from(metadata_entry.digest).0,
-                        &Array4x12::from(cmd.measurement).0,
-                    );
-                    auth_result = AUTHORIZE_IMAGE;
+                &persistent_data.auth_manifest_image_metadata_set;
+
+            // Check if authorization is ignored for the firmware id.
+            for metadata_entry in auth_manifest_image_metadata_col
+                .image_metadata
+                .image_metadata_list
+                .iter()
+            {
+                if cmd.fw_id == metadata_entry.fw_id {
+                    let flags = ImageMetadataFlags(metadata_entry.flags);
+                    if flags.ignore_auth_check() {
+                        return Ok(MailboxResp::AuthorizeAndStash(AuthorizeAndStashResp {
+                            hdr: MailboxRespHeader::default(),
+                            auth_req_result: IMAGE_AUTHORIZED,
+                        }));
+                    }
                     break;
                 }
             }
 
+            let mut auth_result = IMAGE_NOT_AUTHORIZED;
+            let mut fw_id_found = false;
+            for metadata_entry in auth_manifest_image_metadata_col
+                .image_metadata
+                .image_metadata_list
+                .iter()
+            {
+                if cmd.fw_id == metadata_entry.fw_id {
+                    fw_id_found = true;
+                    if cfi_launder(metadata_entry.digest) == cmd.measurement {
```

Comment:
We should use constant-time comparisons for user-provided hashes when possible, e.g., `constant_time_eq::constant_time_eq`.